src/controllers/moment.controller.js

```diff
@@ -187,6 +186,67 @@ const restoreMoment = catchAsync(async (req, res) => {
   });
 });
 
+const importMoments = catchAsync(async (req, res) => {
+  const facebookPosts = await convertFacebookPosts(req.file?.path);
+
+  for (const post of facebookPosts) {
+    await TemporaryMoment.create({
+      ...post,
+      userId: req.user.id,
+    });
+  }
+
+  res.status(httpStatus.OK).json({
+    statusCode: httpStatus.OK,
+    message: i18n.translate('moment.importMomentsSuccess'),
+  });
+});
+
+const getMyTemporaryMoments = catchAsync(async (req, res) => {
+  const temporaryMoments = await TemporaryMoment.find({ userId: req.user.id });
+
+  res.status(httpStatus.OK).json({
+    statusCode: httpStatus.OK,
+    message: i18n.translate('moment.getMyTemporaryMomentsSuccess'),
+    data: {
+      moments: temporaryMoments,
+    },
+  });
+});
+
+const moveMomentToPermanent = catchAsync(async (req, res) => {
+  const { momentIds } = req.body;
+
+  const temporaryMoments = [];
+
+  for (const momentId of momentIds) {
+    const temporaryMoment = await TemporaryMoment.findOne({ $and: [{ _id: momentId }, { userId: req.user.id }] });
+
+    if (!temporaryMoment) {
+      throw new ApiError(httpStatus.NOT_FOUND, i18n.translate('moment.momentNotFound'));
+    }
+
+    temporaryMoments.push(temporaryMoment);
+  }
```

Comment:
tìm hiểu toán tử $in không loop tìm từng cái như thế. thêm 2 biến đếm thành công và thất bài. thành công mỗi bản ghi thì tăng rồi trả về client không nhất thiết trả về lỗi không tìm thấy luôn vì tránh block luồng